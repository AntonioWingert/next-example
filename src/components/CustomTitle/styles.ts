import tw from 'tailwind-styled-components';

import { TitleProps } from '.';
import theme from '../../styles/theme';

export const Container = tw.h1<TitleProps>`
${({ $color }) => ($color ? theme.colors[$color] : theme.colors.primary)}
${({ $size }) => ($size ? theme['text-sizes'][$size] : theme['text-sizes'].md)}

`;
