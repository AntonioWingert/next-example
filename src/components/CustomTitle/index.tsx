import React from 'react'

import * as S from './styles'

export type TitleProps = {
  children: React.ReactNode
  $size?: 'sm' | 'md' | 'lg'
  $color?: 'primary' | 'secondary' | 'danger'
  $as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'
}

export default function CustomTitle({
  children,
  $size = 'md',
  $color = 'primary',
  $as = 'h1',
}: TitleProps) {
  return (
    <S.Container $color={$color} $size={$size} $as={$as}>
      {children}
    </S.Container>
  )
}
