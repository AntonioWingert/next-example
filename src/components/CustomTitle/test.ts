import { render, screen } from '@testing-library/react';
import CustomTitle from '.';

describe('<CustomTitle />', () => {
  it('should render the button with all props passed', () => {
    const { container } = render(
      CustomTitle({
        children: 'Test',
        $as: 'h1',
        $size: 'lg',
        $color: 'primary',
      }),
    );

    expect(screen.getByRole('heading', { name: /Test/i })).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the button with default props', () => {
    const { container } = render(CustomTitle({ children: 'Test' }));

    expect(screen.getByRole('heading', { name: /Test/i })).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
