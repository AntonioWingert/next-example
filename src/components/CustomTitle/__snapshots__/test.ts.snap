// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomTitle /> should render the button with all props passed 1`] = `
<h1
  class="text-blue-200 text-6xl"
>
  Test
</h1>
`;

exports[`<CustomTitle /> should render the button with default props 1`] = `
<h1
  class="text-blue-200 text-4xl"
>
  Test
</h1>
`;
